drivers/adc/Kconfig.intel_quark

```diff
@@ -9,12 +9,15 @@
 menuconfig ADC_INTEL_QUARK_D2000
 	bool "ADC Driver for Intel Quark D2000"
 	depends on SOC_QUARK_D2000
-	select ADC_0
 	help
 	  Enable the ADC driver implementation of the Intel Quark D2000.
 
 if ADC_INTEL_QUARK_D2000
 
+config ADC_0
+	bool "Enable ADC 0"
+	default y
```

Comment:
I'm wondering, will there be a use case for not enabling the only ADC instance while enabling the ADC driver? Maybe the instance selection should be just dropped for the drivers that handle only one ADC instance?

Reply:
Both ways are good for me. I was thinking that someone may enable the ADC driver and forget to enable ADC instances because he thinks there is only one ADC instance. This way no one forgets to enable ADC instance.
Removing this forces people to select instances, so they would have a clear mind about which instances they enabled.

Reply:
> Removing this forces people to select instances, so they would have a clear mind about which instances they enabled.

I am not sure if we understand each other well. What I was trying to propose is not to just remove `default y`, I was thinking about removing the whole `ADC_0` option. My point is that for drivers that handle only one instance this option does not seem to make much sense. It will probably always need to be enabled (either by `default y` or explicitly) anyway. What would be the point of enabling the ADC driver and not enabling any ADC instances?

Reply:
Yes, I agree there's no point to select a driver without selecting an instance. I was thinking about the consistency, if someone has been using ADC with one instance all the time, and the time he's going to start to use a multiple instances ADC. he might forget to choose any instances.
So, how about this, remove 'ADC_0' from every single instance drivers, and add 'default y' to ADC_0 to the ones with multiple instances.

Reply:
> So, how about this, remove 'ADC_0' from every single instance drivers, and add 'default y' to ADC_0 to the ones with multiple instances.

Yes, that's exactly what I was thinking about. Such approach should be convenient for everyone - no need for this a bit artificial `#ifdef CONFIG_ADC_0` in single instance drivers, and in every case some ADC instance enabled, so any sample or test application that just sets `CONFIG_ADC=y` has some device to work with.